### DISCUSSION TITLE
Вопрос по max. размерам панели

### DISCUSSION THREAD

**Discussion by sgt90125** [vvip-68/GyverPanelWiFi#279] [Задавайте вопросы]
Здравствуйте.
В описании проекта написано : ....  (до 128 по ширине и/или высоте).
Это 128 диодов по ширине и 128 диодов по высоте ? Неважно это одна матрица(128х128) или несколько матриц с общим количеством диодов 128х128 (например матриц 32х16 будет 4х8=32). Так? 
   Или же это количество матриц? Тогда это будет действительно "настенные экраны больших размеров".

**Comment by vvip-68** [comment]
128 диодов - это 128 диодов. Не матриц, не сегментов. ДИОДОВ!
Это DIY-проект. Он не предназначен для телевизионных больших экранов, как их собирают из модулей P10.
Контроллер ESP8266 не тянет на этом проекте более 800 диодов. ESP32 вроде тянет порядка 2400 диодов, при бОльшем количестве - начинаются проблемы. Для настольных часов и оконных новогодних гирлянд этих размеров вполне достаточно. Для огромных экранов а-ля телевизор - вам в другой проект.

**Comment by sgt90125** [comment]
Спасибо за ответ.
128 при шаге 10 это всего лишь 1 метр с небольшим.
Хотелось бы побольше .
А Вы бы не взялись сделать проект на большие панели на ESP32?
С возможностью подключения их (ESP32) к стандартным китайским матрицам P10 или с ещё более плотной упаковкой P8, P5,P4.  
  
>Суббота, 20 мая 2023, 0:23 +05:00 от vvip-68 ***@***.***>:
> 
> 
>128 диодов - это 128 диодов. Не матриц, не сегментов. ДИОДОВ!
>Это DIY-проект. Он не предназначен для телевизионных больших экранов, как их собирают из модулей P10.
>Контроллер ESP8266 не тянет на этом проекте более 800 диодов. ESP32 вроде тянет порядка 2400 диодов, при бОльшем количестве - начинаются проблемы. Для настольных часов и оконных новогодних гирлянд этих размеров вполне достаточно. Для огромных экранов а-ля телевизор - вам в другой проект.
>—
>Reply to this email directly,  view it on GitHub , or  unsubscribe .
>You are receiving this because you authored the thread. Message ID:  <vvip-68/GyverPanelWiFi/repo-discussions/279/comments/5951886 @ github . com> 
 
 
--
number six
Отправлено из Почты  Mail.ru

**Reply by vvip-68** [reply]
128x128x3 - это уже 50К памяти. В среднем для работы для различных буферов требуется еще почти столько же памяти. Это уже 100К оперативы надо. В ESP32 ее свободной остается после системных расходов порядка 170К и, как показала практика, используемые библиотеки и этот объем не очень хотят/умеют использовать - чего-то валятся.

Так что ESP32 не потянет реально большие экраны. Тут уже нужно на Raspberry смотреть. Это не мое. Я чистый хоббийщик, которому пару-тройку лет назад понравилась лампа Гайвера и он решил кое что переделать "под себя".

Далее. Уже 128x128 - это более 16 тыс диодов. Стандартная адресная лента WS2812 с ее 800кГц тактовой частотой вывод кадра - примерно 1 кадр в 2 секунды. Даже если распараллелить выводы - будет не быстрее 2 кадров в секунду. На "телевизор" явно не тянет. WS8213 имеет в 2.5 раза больше заявленную скорость - но все равно будет маловато, даже на 128x128...

Проектов на матрицы P10 в инете, в том числе и на гитхабе - как грязи. На любой вкус. И на ESP32 и на распберри. Но все равно все что я видел - максимально, насколько мне помнится - 128x128 точек или того меньше. Что я могу сделать лучше чем уже есть и зачем? ESP32 - не такой уж мощный микроконтроллер и для реально больших проектов не годится.

**Comment by sgt90125** [comment]
Хочу поблагодарить за отличную разработку. Я её повторил. Повесил у себя над входом в инетклуб. По ссылке  https://cloud.mail.ru/public/zPLe/DinwbyMwa мои попытки передавать скрины экрана игр через Jinx. Лента 60 на метр. Получается где-то P17.Тут у нас не мог найти 144 на метр. С али ждать не хотел (2 месяца).На ESP 8266 NodeMCU lolin.
Ещё раз спасибо.
P.S. Наглость конечно, но хотел вас спросить, не сделаете разработку для управлению розеткой через телеграм? Такие есть согласен. Но нужно чтобы настройки Wi-fi, chatID и токен бота можно было менять в приложении со смартфона.
 
--
number six
Отправлено из Почты  Mail.ru
